shared/actions/gregor.js

```diff
@@ -110,7 +110,8 @@ function registerGregorListeners() {
 }
 
 function* handleTLFUpdate(items: Array<Types.NonNullGregorItem>): SagaGenerator<any, any> {
-  const seenMsgs: Types.MsgMap = yield Saga.select((state: TypedState) => state.gregor.seenMsgs)
+  const state: TypedState = yield Saga.select()
```

Comment:
This could be pure

Reply:
You'd just change the below to be:
```

function handlePushState(action:..., state) {
...
return Saga.sequentially([
  handleTLFUpdate(nonNullItems, state)
  ...
]) 
```

Reply:
(assuming you don't need to see the latest state)

Reply:
i didn't cause this is actually called by another saga below. and while i could make that work that'd be less of a simple refactor so i avoided it

Reply:
(which is your suggestion), sorry i got confused about it. still my goal isn't to apply pure to 100% of the things. just the simplest stuff. hopefully with the explorations in devices (the discussion one) we'd eliminate a bunch of this type of code anyways